cmake/findDependencies.cmake

```diff
@@ -7,12 +7,19 @@ if (BUILD_GUI)
 endif()
 
 if (HAVE_RULES)
-    find_library(PCRE pcre)
-    if (NOT PCRE)
+    find_library(PCRE_LIBRARY pcre)
+    if (NOT PCRE_LIBRARY)
         message(FATAL_ERROR "pcre dependency for RULES has not been found")
     endif()
 endif()
 
+if (USE_Z3)
+    find_library(Z3_LIBRARY z3)
```

Comment:
This should do `find_package(Z3)`.

Reply:
CMake doesn't provide a `FindZ3.cmake` module so we cannot do that.

Reply:
No need for `FindZ3.cmake` as Z3 provides `Z3Config.cmake` when installing(at least on the latest version).

Reply:
Interesting - I am not aware how this configs work. Will look into this when trying to get this z3 version thing figured out. We should do the same for pcre.

Reply:
pcre doesn't provide PCREConfig.cmake so its not possible.

Reply:
So the `Z3Config.cmake` file was provided starting with version 4.6.0. Do we want to support versions older than that in the cmake? Its easy enough to install newer version using cget(ie `cget -p /usr/local install Z3Prover/z3@z3-4.8.8`), so dont see why thats necessary.

Reply:
Debian comes with 4.4 and the oldest platform to support out-of-the-box is CentOS 7. Will check the external stuff with that and do adjustments based on that.